### PR TITLE
feat: configurable microsoft apt repository

### LINKED ISSUE #3677: configurable packages.microsoft.com apt repository
Some environments might not have access to `packages.microsoft.com`, let's make that apt repository reference configurable.

### PR DESCRIPTION
**Pull request by jackfrancis**
**Reason for Change**:

This PR enables configurable override of the default `packages.microsoft.com` apt repository for sourcing Ubuntu 16.04-LTS and Ubuntu 18.04-LTS packages.

**Issue Fixed**:

Fixes #3677

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by jackfrancis** [comment]
@DavidParks8 FYI

File: pkg/api/defaults_test.go [hunk 1]
```diff
@@ -914,6 +914,30 @@ func TestEtcdStorageLimitGB(t *testing.T) {
...
 }
 
+func TestMicrosoftAptRepositoryURL(t *testing.T) {
+	mockCS := getMockBaseContainerService("1.18.5")
+	properties := mockCS.Properties
+	properties.OrchestratorProfile.OrchestratorType = Kubernetes
+	properties.MasterProfile.Count = 1
+	mockCS.setOrchestratorDefaults(false, false)
+	if properties.OrchestratorProfile.KubernetesConfig.MicrosoftAptRepositoryURL != DefaultMicrosoftAptRepositoryURL {
+		t.Fatalf("MicrosoftAptRepositoryURL did not have the expected size, got %s, expected %s",
```
Comment (mboersma) [inline pkg/api/defaults_test.go:924]:
```suggestion
		t.Fatalf("MicrosoftAptRepositoryURL did not have the expected contents, got %s, expected %s",
```

File: pkg/api/defaults_test.go [hunk 2]
```diff
@@ -914,6 +914,30 @@ func TestEtcdStorageLimitGB(t *testing.T) {
...
+
+	// validate defaults doesn't override valid value
+	mockCS = getMockBaseContainerService("1.18.5")
+	properties = mockCS.Properties
+	properties.OrchestratorProfile.OrchestratorType = Kubernetes
+	properties.MasterProfile.Count = 1
+	properties.OrchestratorProfile.KubernetesConfig.MicrosoftAptRepositoryURL = "custom.packages.com"
+	mockCS.setOrchestratorDefaults(false, false)
+	if properties.OrchestratorProfile.KubernetesConfig.MicrosoftAptRepositoryURL != "custom.packages.com" {
+		t.Fatalf("MicrosoftAptRepositoryURL did not have the expected size, got %s, expected custom.packages.com",
```
Comment (mboersma) [inline pkg/api/defaults_test.go:936]:
```suggestion
		t.Fatalf("MicrosoftAptRepositoryURL did not have the expected contents, got %s, expected custom.packages.com",
```

File: test/e2e/test_cluster_configs/everything.json [hunk 3]
```diff
@@ -89,7 +89,8 @@
 					"loadBalancerOutboundIPs": 2,
 					"apiServerConfig": {
 						"--anonymous-auth": "true"
-					}
+					},
+					"microsoftAptRepositoryURL": "https://packages.microsoft.com"
```
Comment (mboersma) [inline test/e2e/test_cluster_configs/everything.json:93]: Are you being explicit about the default here for documentation reasons?

**Pull request merged**